lib/hexpm_web/controllers/web_auth_controller.ex

```diff
@@ -0,0 +1,54 @@
+defmodule HexpmWeb.WebAuthController do
+  use HexpmWeb, :controller
+  @moduledoc false
+
+  alias Hexpm.Accounts.WebAuth
+
+  # Controller for Web Auth, a means of authenticating the cli from the website
+
+  plug :requires_login when action == :show
+
+  # step one of device flow
+  def code(conn, %{"scope" => scope}) do
+    case WebAuth.get_code(scope) do
+      {:error, _changeset} ->
+        conn
+        |> put_status(:unprocessable_entity)
+        |> json(%{"error" => "invalid scope"})
+
+      response ->
+        json(conn, response)
```

Comment:
This should be rendered in a view `WebAuthView` and explicit select which fields to render so that we do not accidentally include private data if `WebAuth.get_code/1` returns more information in the future.

Same for the `success` and `access_token` actions.